src/core/network/qgsblockingnetworkrequest.h

```diff
@@ -207,6 +223,12 @@ class CORE_EXPORT QgsBlockingNetworkRequest : public QObject
      */
     void downloadFinished();
 
+    /**
+     * Emitted when when data are sent during a request.
+     * \since QGIS 3.22
+     */
+    void uploadProgress( qint64, qint64 );
```

Comment:
Because *downloadFinished* is already fired when a PUT request is done. I can change this and emit an *uploadFinished* on a PUT request instead but it's part of the API, some users may have already connected to this signal.

Reply:
I am not sure what is best:

1) add a new signal `uploadFinished ` (and removing the emission of `downloadFinished` with QGIS 4) sounds idela but it's almost impossible to identify for developpers (no deprecation).
2) create a new signal `finished` emitted in both scenarios, deprecate `downloadFinished `
3) stick to the current implementation aka use a signal with an imprecise phrasing.

My preference would go to 2, but I have no strong opinion.

Reply:
I vote also for solution 2. Done